src/test/groovy/sdkman/support/UnixUtils.groovy

```diff
@@ -3,10 +3,14 @@ package sdkman.support
 class UnixUtils {
 
 	static getPlatform() {
-		asSdkmanPlatform(System.getProperty("os.name"))
+		asSdkmanPlatform(System.getProperty("os.name"), System.getProperty("os.arch"))
 	}
 
-	static asSdkmanPlatform(platform) {
+	static asSdkmanPlatform(platform, architecture = null) {
```

Comment:
Having a `null` default doesn't feel right. You could just leave it off as it will always be set.

Reply:
This method is also used in the cucumber steps with only 1 parameter. Without the default value, we either need to create another overloaded method with the default architecture or change all the places where this method is used to provide the 2 parameters. Thought this was the easiest way (least amount of code) to solve it now. What do you think?

Reply:
I'd prefer to always pass it in explicitly if possible, even if it means updating the cucumber features.

Reply:
No problem, I have change them in the steps